tests/QrReaderTest.php

```diff
@@ -2,16 +2,17 @@
 
 namespace Khanamiryan\QrCodeTests;
 
+use PHPUnit\Framework\TestCase;
 use Zxing\QrReader;
 
-class QrReaderTest extends \PHPUnit\Framework\TestCase
+class QrReaderTest extends TestCase
 {
 
     public function testText1()
     {
         $image = __DIR__ . "/qrcodes/hello_world.png";
 
         $qrcode = new QrReader($image);
-        $this->assertSame("Hello world!", $qrcode->text());
+        self::assertSame("Hello world!", $qrcode->text());
```

Comment:
This function isn't static, you should still use `$this->` 🤔

Reply:
Please take a look here: https://github.com/sebastianbergmann/phpunit/blob/1f0325a6d67a052df841de667ec6f1f3b19b3337/src/Framework/Assert.php#L1327 :-)

Reply:
Yes but that's just so you can call it statically if the method you're calling it from is also static. It's conventional to always use `$this->` for tests: https://phpunit.readthedocs.io/en/9.3/assertions.html#assertsame

Reply:
I found this article by Sebastian Bergmann where he writes, that there's no right or wrong about that: https://phpunit.de/manual/6.5/en/appendixes.assertions.html#appendixes.assertions.static-vs-non-static-usage-of-assertion-methods I prefer calling static methods statically, but I don't want to waste our time about that topic and changed it back. :-)